### PR TITLE
comp: add alert and button

### PR DESCRIPTION
**Pull request by PuneetP16**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -28,17 +28,19 @@ <h1 class="h4 nav_head">Get Started</h1>
 			<h1 class="h4 nav_head">Components</h1>
 			<ul class="nav_items_list flex-col-align-center">
 				<li class="nav_items">
-					<a
-						class="nav_links nav_link-selected"
-						href="/src/components/avatar/avatar.html"
+					<a class="nav_links nav_link-selected" href="/src/components/avatar/avatar.html"
 						>Avatar</a
 					>
```
Comment (manojsarna) [inline index.html:35]: check the alignment of the tags

File: index.html [hunk 2]
```diff
@@ -28,17 +28,19 @@ <h1 class="h4 nav_head">Get Started</h1>
...
 				<li class="nav_items">
 					<a class="nav_links" href="/src/components/badge/badge.html">Badge</a>
 				</li>
 				<li class="nav_items">
-					<a class="nav_links" href="/src/components/alert/alert.html">Alert</a>
+					<a
+						class="nav_links"
+						href="/src/components/alert/alert.html"
+						>Alert</a
+					>
```
Comment (manojsarna) [inline index.html:45]: check here too

File: index.html [hunk 3]
```diff
@@ -49,13 +51,13 @@ <h1 class="h4 nav_head">Components</h1>
...
 				</li>
 				<li class="nav_items">
-					<a class="nav_links" href="/src/components/card/card.html">Form</a>
+					<a class="nav_links" href="/src/components/form/form.html">Form</a>
 				</li>
 				<li class="nav_items">
-					<a class="nav_links" href="/src/components/card/card.html"
+					<a class="nav_links" href="/src/components/snackbar/snackbar.html"
 						>Snackbar</a
 					>
```
Comment (manojsarna) [inline index.html:66]: check here too

**Review by manojsarna — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (PuneetP16) [reply index.html:35]: It was auto-formatted by vs-code exetension. I will look into it. Thanks for pointing it.

**Comment by m26unkwn** [comment]
Love the UI brother Buttons are so clean.

**Review by m26unkwn — APPROVED** [review]

File: src/components/alert/alert.html [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+			<div class="component_display_container flex-center margin-top">
+				<div class="alerts flex-col-align-center fw">
+					<div id="alert--warning--dismiss" class="alert alert--dismissable">
+						This is a dismissable warning alert! (add width:fit-content to
+						reduce the size of the alert)
+						<button
+							id="alert--warning--dismiss-btn"
+							class="btn btn--icon btn--close--transparent alert--btn__dismiss btn--circular"
+						>
+							<i class="bx bx-x"></i>
```
Comment (saurabhchirde) [inline src/components/alert/alert.html:189]: Confusing naming of 'bx' , 'bx-x'
Can improve these.

**Review by saurabhchirde — APPROVED** [review]
Looks good to merge.

File: src/components/alert/alert.css [hunk 5]
```diff
@@ -0,0 +1,68 @@
+.alerts {
+	gap: 0.8rem;
```
Comment (iRohitGaur) [inline src/components/alert/alert.css]: use px for values less than 10px

File: src/components/alert/alert.html [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+			<div class="snippet_container">
+				<script
+					type="text/plain"
+					class="line-numbers language-css snippet_container__script"
+				>
+				//Dismiss Alert
+				<div id="alert--warning--dismiss" class="alert alert--dismissable">
+				  <i class='bx bx-bug' ></i>
+				  //Your Alert text goes here..
+				  <button id="alert--warning--dismiss-btn" class="btn--icon alert--btn__dismiss">
```
Comment (iRohitGaur) [inline src/components/alert/alert.html]: consistency in class names required

File: src/components/badge/badge.html [hunk 7]
```diff
@@ -124,16 +130,38 @@ <h3 class="mt1p5">Badge with Text</h3>
...
+					type="text/plain"
+					class="line-numbers language-css snippet_container__script"
+				>
+					//In parent tag add badge_base class and in child tag add badge class
+					//  with color, size and position as shown below
+
+					//large badge
+					<span class="badge_base h2">
+						//Text Goes Here
+						<span class="badge color-primary-bg w-h-lh-lg position-right-top-lg">
```
Comment (iRohitGaur) [inline src/components/badge/badge.html]: give meaningful names. "w-h-lh-lg" is confusing

File: src/components/badge/badge.html [hunk 8]
```diff
@@ -142,7 +170,15 @@ <h3 class="mt1p5">Badge with Button</h3>
...
 				<div class="badges flex-col-align-center">
 					<span class="h4">Button with a large badge on the left</span>
-					<button class="button_primary badge_base color-primary-bg">
+					<!-- <button class="button--primary badge_base color-primary-bg">
+						Action
+						<span
+							class="badge color-secondary-bg w-h-lh-lg position-left-top-lg"
+							>8</span
+						>
+					</button> -->
```
Comment (iRohitGaur) [inline src/components/badge/badge.html]: remove commented code

File: src/components/button/button.css [hunk 9]
```diff
@@ -1,17 +1,221 @@
...
+
+.btn {
+	border-radius: 5px;
+	box-shadow: 0 3px 1px -2px #0003, 0 2px 2px #00000024, 0 1px 5px #0000001f;
 	cursor: pointer;
-	padding: 0.5rem 1rem;
+	display: inline-block;
+	font-weight: 600;
+	opacity: 0.9;
+	padding:1rem;
```
Comment (iRohitGaur) [inline src/components/button/button.css]: give spacing after :

File: src/components/button/button.css [hunk 10]
```diff
@@ -1,17 +1,221 @@
...
+.btn--primary-light {
+	background-color: var(--COLOR-PRIMARY-DARK);
+	border: 1px solid var(--COLOR-PRIMARY-DARK);
+	color: white;
+}
+
+.transparent--black {
+	background-color: white;
+	color: black;
+	/* border: white; */
```
Comment (iRohitGaur) [inline src/components/button/button.css]: remove commented code

File: src/components/button/button.css [hunk 11]
```diff
@@ -1,17 +1,221 @@
...
+	border: none;
+	box-shadow: 0 3px 1px -2px #0003, 0 2px 2px #00000024, 0 1px 5px #0000001f;
+	cursor: pointer;
+	font-weight: 500;
+	padding: 0.5rem 1rem;
 	text-align: center;
+}
+
+.bx--shadow {
 	box-shadow: 0 3px 1px -2px #0003, 0 2px 2px #00000024, 0 1px 5px #0000001f;
```
Comment (iRohitGaur) [inline src/components/button/button.css]: use variables for color

**Review by iRohitGaur — APPROVED** [review]
Nice work! Good to merge after changes 👍🏻

File: src/components/alert/alert.html (hunk 4 above)
Comment (PuneetP16) [reply src/components/alert/alert.html:189]: These are of BOXICON font icons.

File: src/components/alert/alert.html [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+				</span>
+			</p>
+
+			<!-- component display container -->
+			<div class="component_display_container flex-center margin-top">
+				<div class="alerts flex-col-align-center fw">
+					<div id="alert--warning--dismiss" class="alert alert--dismissable">
+						This is a dismissable warning alert! (add width:fit-content to
+						reduce the size of the alert)
+						<button
```
Comment (pankaj-prog) [inline src/components/alert/alert.html:185]: dismiss button is not visible.

**Review by pankaj-prog — APPROVED** [review]
UI looks good.

File: src/components/avatar/avatar.html [hunk 13]
```diff
@@ -185,11 +207,13 @@ <h3>Text Avatars</h3>
 				include class avatar-text along with avatar and size class.
 
 				<br />
-				<span class="m_doc_classes">avatar__text</span>
-				<span class="m_doc_classes">avatar--lg</span> with
-				<span class="m_doc_classes">avatar__text--lg</span> and
-				<span class="m_doc_classes">avatar--sm</span> with
-				<span class="m_doc_classes">avatar__text--sm</span>
+				<span class="m_doc_class">
```
Comment (tarunsankhla) [inline src/components/avatar/avatar.html]: Can there be consistency - and  _

File: src/components/button/button.css [hunk 14]
```diff
@@ -1,17 +1,221 @@
...
+.btn--icon i {
+	font-size: 2.4rem;
+}
+
+.btn--icon {
+	align-items: center;
+	border-radius: 5;
+	cursor: pointer;
+	display: flex;
+	gap: 0.8rem;
```
Comment (tarunsankhla) [inline src/components/button/button.css]: Gap should be in px if screen enlarge more gap

**Comment by j24m** [comment]
Hello, You can make the changes suggested by others , like consistency in spacing and a bit understandable naming convention.
UI as its name looks magnificient. Great work:)

File: src/components/alert/alert.html (hunk 12 above)
Comment (PuneetP16) [reply src/components/alert/alert.html:185]: It was visible, check again.

**Comment by PuneetP16** [comment]
Thanks for all of your suggestions, I have committed changes that will reflect the fixes.

**Pull request merged**